runtime/src/accounts.rs

```diff
@@ -262,14 +262,20 @@ impl Accounts {
     }
 
     /// Slow because lock is held for 1 operation instead of many
-    pub fn load_slow(
-        &self,
-        ancestors: &HashMap<Fork, usize>,
-        pubkey: &Pubkey,
-    ) -> Option<(Account, Fork)> {
-        self.accounts_db
+    pub fn load_slow(&self, ancestors: &HashMap<Fork, usize>, pubkey: &Pubkey) -> Option<Account> {
+        let mut account = self
+            .accounts_db
             .load_slow(ancestors, pubkey)
-            .filter(|(acc, _)| acc.lamports != 0)
+            .unwrap_or_default()
+            .0;
+
+        account.lamports += self.credit_only_pending_credits(pubkey);
```

Comment:
ack more locks

Reply:
I guess it's the _slow path though.

Reply:
only for RPC or tests...

Reply:
we have a bunch of RPC clients that fail if they issue a credit-only transfer because they immediately query balance, which doesn't get updated until the end of the slot...

Reply:
Once @CriesofCarrots completes #6346 this ought to get much better, as the clients should really wait for the cluster to vote